turbinia/lib/llm_libs/vertex_ai_lib.py

```diff
@@ -0,0 +1,78 @@
+"""Turbinia LLM library that uses VertexAI APIs."""
+
+from __future__ import unicode_literals
+
+import logging
+
+import google.generativeai as genai
+from turbinia import config as turbinia_config
+from turbinia.lib.llm_libs import llm_lib_base
+
+MODEL_NAME = "gemini-1.0-pro"
+MAX_OUTOUT_TOKEN = 2048
+MODEL_TEMPRATURE = 0.2
+# GCP_GENERATIVE_LANGUAGE_API_KEY must be defined in turbinia_config
+GENERATIVE_CONFIG = {
+    "temperature": MODEL_TEMPRATURE,
+    "max_output_tokens": MAX_OUTOUT_TOKEN,
+}
+SAFETY_SETTINGS = [
+    {
+        "category": "HARM_CATEGORY_DANGEROUS",
+        "threshold": "BLOCK_NONE",
+    },
+    {
+        "category": "HARM_CATEGORY_HARASSMENT",
+        "threshold": "BLOCK_NONE",
+    },
+    {
+        "category": "HARM_CATEGORY_HATE_SPEECH",
+        "threshold": "BLOCK_NONE",
+    },
+    {
+        "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
+        "threshold": "BLOCK_NONE",
+    },
+    {
+        "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
+        "threshold": "BLOCK_NONE",
+    },
+]
+log = logging.getLogger("turbinia")
+
+
+class TurbiniaVertexAILib(llm_lib_base.TurbiniaLLMLibBase):
+  """Turbinia LLM library that uses VertexAI APIs."""
+
+  def prompt_with_history(
+      self, prompt_text: str, history_session: genai.ChatSession = None) -> str:
+    """Sends a prompt to the Gemini-pro model using VertexAI.
+
+    Args:
+        prompt_text: The text of the prompt.
+        history_session: optional conversation history if it is desired to keep
+        the state of the conversation, i.e. a chat.
+
+    Returns:
+        A tuple of the response from the Gemini-pro model and a history session.
+    """
+    log.info('Calling VertexAI using generative model "%s"', MODEL_NAME)
+    genai.configure(api_key=turbinia_config.GCP_GENERATIVE_LANGUAGE_API_KEY)
```

Comment:
How about checking if the API_KEY is indeed set or not. If not set, generate a warning message and return to prevent errorring out if no API is set.